pycloudlib/azure/instance.py

```diff
@@ -136,13 +137,27 @@ def _do_restart(self, **kwargs):
             resource_group_name=self._instance["rg_name"], vm_name=self.name
         )
 
-    def delete(self, wait=True):
+    # pylint: disable=broad-except
+    def delete(self, wait=True) -> List[Exception]:
         """Delete instance."""
-        delete = self._client.virtual_machines.begin_delete(
-            resource_group_name=self._instance["rg_name"], vm_name=self.name
-        )
-
-        if wait:
-            delete.wait()
+        if self.status == "deleted":
+            return []
+        try:
+            poller = self._client.virtual_machines.begin_delete(
```

Comment:
What if we delete the resource group instead of the vm, since it will delete every other resource allocated in the same resource group?